Jahsie/main.js

```diff
@@ -0,0 +1,66 @@
+// Your card and deque factory code should be loaded into the browser before this file
+// by the main.html container file.
+// At this point, both makeCard and makeDeque should be defined.
+
+
+// 2b:
+// make a deque instance to store a full deck of cards:
+var deckOfCards = makeDeque(makeCard.fullSet);
+
+ascendingSuit = function(x,y) {
+    var n = y.suit() - x.suit();
+    if (n !== 0) {
+        return n;
+     }
+     return y.rank() - x.rank();
+      };
+
+alphabeticalSort = function(x,y) {
+	if (x.name() > y.name()) return -1;
+	   else if (x.name() < y.name()) return 1;
+	   else return 0;
+};
+
+
+deckOfCards.sort(ascendingSuit);
+deckOfCards.cut();
+console.log("For some reason when I call deckOfCards.cut() I wind up losing half of makeCard.fullset.  Other than that the cut method works perfectly, and when I tested the makeDeque.cut on other arrays I never ran in to the same problem. Any ideas what is causing this?");
```

Comment:
hmm, interesting. On the up side, the values in `deckOfCards` stick around and cut properly, which is where you want the data to be. I'll have a closer look at this and see if I can suss out what's going on.